docs/design/UseOfOptional.md

```diff
@@ -2,19 +2,15 @@
 
 ## Use of Optional
 
-This page describes general guidelines of how we use
-`java.util.Optional`.
+This page describes a general guideline for the use of
+`java.util.Optional` in the AWS SDK for Java 2.x.
 
-- `Optional` should be used when it isn't obvious to a caller whether a
-  result will be null.
-- `Optional` must not be used when the result will never be
-  `Optional.empty()`.
-- `Optional` must not be used for instance (in `@SdkPublicApi`s).
-- `Optional` must not be used for parameters (in `@SdkPublicApi`s).
-- `Optional` must not be used when the caller may know whether a value
-  is present. As a consequence:
-  - `Optional` must not be used in classes with "setters" and "getters",
-    like Builders or POJOs.
+- `Optional` SHOULD be used when it is not obvious to a caller whether a
+  result will be null, e.g, `public <T> Optional<T> getValueForField(String fieldName, Class<T> clazz)`in [SdkResponse.java](https://github.com/aws/aws-sdk-java-v2/blob/aa161c564c580ced4a0381d3ed7d4d13120916fc/core/sdk-core/src/main/java/software/amazon/awssdk/core/SdkResponse.java#L59-L61)
+- `Optional` MUST NOT be used when the result will never be null.
+- `Optional` SHOULD NOT be used for member variables, e.g., `private final Optional<String> field;`
+- `Optional` MUST NOT be used for method parameters, e.g., `private void test(Optional<String> value)`
+- `Optional` MUST NOT be used as a return type for "getters" in generated service model classes such as service Builders or POJOs.
```

Comment:
I wonder if we should separate these rules into categories:
1. Return types
2. Method parameters
3. Member variables (or just _other_)

Currently, your proposed \1 and \2 seem to refer to return types, but it doesn't explicitly state that. And then we jump back and forth a bit before returning to return types.

Reply:
Good point, will clarify